src/main/java/ru/ayaz/UserMessageHandler.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    public void addWriterToMap(String nickname, PrintWriter writer) {
+        outputWriterMap.put(nickname, writer);
+    }
+
+    public void addReaderToMap(String nickname, BufferedReader reader) {
+        inputReaderMap.put(nickname, reader);
+    }
+
+
+    @Override
+    public void run() {
+        try {
+            while (true) {
+                UserMessage message = takeMessage();
+                if (parser.isCommand(message.getText())) {
+                    executeCommand(message);
+                } else {
+                    sendMessageToEveryone(message);
+                }
+            }
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    private void executeCommand(UserMessage commandMessage) {
+        String nickname = commandMessage.getSenderName();
+        String text = commandMessage.getText();
+
+        switch (parser.getCommand(text)) {
+            case "#quit":
+                executeQuitCommand(commandMessage);
+                break;
+            case "#ignore":
+                executeIgnoreCommand(commandMessage);
+                break;
+            default:
+                PrintWriter writer = outputWriterMap.get(nickname);
+                sendMessageToExactPerson("Unknown command", writer);
+                break;
+        }
+    }
+
+
+    private void executeQuitCommand(UserMessage commandMessage) {
+        closeWriterAndReaderInMap(commandMessage.getSenderName());
+    }
+
+
+    private void executeIgnoreCommand(UserMessage commandMessage) {
+        String nickname = commandMessage.getSenderName();
+        String text = commandMessage.getText();
+
+        User user = usersMap.get(nickname);
+        try {
+            user.ignoreUser(parser.getFirstArgument(text));
+        } catch (InvalidUserCommandException e) {
+            PrintWriter writer = outputWriterMap.get(nickname);
+            sendMessageToExactPerson("Invalid argument", writer);
+        }
+    }
+
+
+    private void closeWriterAndReaderInMap(String nickname) {
+        try {
+            outputWriterMap.get(nickname).close();
+            inputReaderMap.get(nickname).close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    private void sendMessageToEveryone(UserMessage message) {
+        String senderNickname = message.getSenderName();
+        String text = message.getText();
+
+        for (Map.Entry<String, User> entry : usersMap.entrySet()) {
+            String receiverNickname = entry.getKey();
+            User user = entry.getValue();
+
+            if (user.isItMe(senderNickname)) {
+                continue;
+            }
+
+            PrintWriter writer = outputWriterMap.get(receiverNickname);
+            sendMessageToExactPerson(senderNickname + ": " + text, writer);
+        }
+    }
+
+
+    private void sendMessageToExactPerson(String text, PrintWriter writer) {
```

Comment:
&#x1F34E; **Issue**
This method could be named better.
You are mixing layers of abstraction here. You say as if you are sending a message to a person. If so, then one will expect the argument to be a person and not a PrintWriter.
Decide, whether you want to be it on higher or lower level of abstraction.